### PR TITLE
Jane Street stable release v1.0

### PR DESCRIPTION
**Pull request by diml**

### CODE REVIEW

**Comment by avsm** [comment]
```
#=== ERROR while installing ppx_driver.v1.0 ===================================#
# opam-version         1.2.2 (58ef3b8213100953848d362f7120a30356d7f77d)
# os                   linux
# command              jbuilder build --only-packages ppx_driver --root . -j 2 @install
# path                 /home/opam/.opam/4.03.0/build/ppx_driver.v1.0
# compiler             4.03.0
# exit-code            1
# env-file             /home/opam/.opam/4.03.0/build/ppx_driver.v1.0/ppx_driver-29-9a7adc.env
# stdout-file          /home/opam/.opam/4.03.0/build/ppx_driver.v1.0/ppx_driver-29-9a7adc.out
# stderr-file          /home/opam/.opam/4.03.0/build/ppx_driver.v1.0/ppx_driver-29-9a7adc.err
### stdout ###
# Actual targets:
# - alias install
### stderr ###
# Workspace root: .
# Running[0]: /home/opam/.opam/4.03.0/bin/ocamlc.opt -config > /tmp/jbuild8412e9.output
# Running[1]: /home/opam/.opam/4.03.0/bin/ocamlfind printconf path > /tmp/jbuildaa0a01.output
# Running[2]: /usr/local/bin/opam config var lib > /tmp/jbuildbd381f.output
# Error: Findlib package "ocaml-migrate-parsetree.driver" not found.
# Hint: try 'jbuilder external-lib-deps --missing'
```

File: packages/async/async.v1.0/descr [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Monadic concurrency library
+
+Part of Jane Street\226\128\153s Core library
```
Comment (avsm) [inline packages/async/async.v1.0/descr]: Do you intend the fancy apostrophe encoding above?

Comment (diml) [inline packages/async/async.v1.0/descr]: No, indeed. I'm adding a check that descriptions are ASCII only

**Comment by avsm** [comment]
Almost there -- just async_js and incr_dom failing now:

```
#=== ERROR while installing js_of_ocaml.2.8.4 =================================#
# opam-version         1.2.2 (58ef3b8213100953848d362f7120a30356d7f77d)
# os                   linux
# command              make build
# path                 /home/opam/.opam/4.03.0/build/js_of_ocaml.2.8.4
# compiler             4.03.0
# exit-code            2
# env-file             /home/opam/.opam/4.03.0/build/js_of_ocaml.2.8.4/js_of_ocaml-29-a49717.env
# stdout-file          /home/opam/.opam/4.03.0/build/js_of_ocaml.2.8.4/js_of_ocaml-29-a49717.out
# stderr-file          /home/opam/.opam/4.03.0/build/js_of_ocaml.2.8.4/js_of_ocaml-29-a49717.err
### stdout ###
# [...]
# ocamlfind ocamlc     -w +A-4-7-9-37-38-41-44-45 -package lwt,uchar -I deriving_json -safe-string -c deriving_json/deriving_Json.mli
# ocamlfind ocamlc     -w +A-4-7-9-37-38-41-44-45 -safe-string -I deriving_json -c deriving_json/deriving_Json.ml
# ocamlfind ocamlc     -w +A-4-7-9-37-38-41-44-45 -a -o deriving_json.cma deriving_json/deriving_Json_lexer.cmo deriving_json/deriving_Json.cmo
# ocamlfind ocamlc     -w +A-4-7-9-37-38-41-44-45 -safe-string -package lwt,graphics -I ./ graphics/graphics_js.mli -o graphics/graphics_js.cmi
# ocamlfind ocamlc     -w +A-4-7-9-37-38-41-44-45 -syntax camlp4o -safe-string -package lwt,lwt.syntax,graphics -ppopt syntax/pa_js.cmo -I graphics -c graphics/graphics_js.ml -o graphics/graphics_js.cmo
# ocamlfind ocamlc     -w +A-4-7-9-37-38-41-44-45 -a -o graphics/graphics.cma graphics/graphics_js.cmo
# ocamlfind ocamlc     -w +A-4-7-9-37-38-41-44-45 -syntax camlp4o -safe-string -package lwt.syntax,async_kernel -ppopt syntax/pa_js.cmo -I . -I async async/async_js.mli -o async/async_js.cmi
# Makefile:157: recipe for target 'async/async_js.cmi' failed
# make[1]: Leaving directory '/home/opam/.opam/4.03.0/build/js_of_ocaml.2.8.4/lib'
# Makefile:15: recipe for target 'library' failed
### stderr ###
# [...]
# File "syntax/pa_js.ml", line 43, characters 7-23:
# Warning 3: deprecated: String.lowercase
# Use String.lowercase_ascii instead.
# File "syntax/pa_js.ml", line 43, characters 7-23:
# Warning 3: deprecated: String.lowercase
# Use String.lowercase_ascii instead.
# File "async/async_js.mli", line 22, characters 5-21:
# Error: Unbound module Async_kernel.Std
# make[1]: *** [async/async_js.cmi] Error 2
# make: *** [library] Error 2
```

do we need a release of js_of_ocaml to match the Async release, @hhugo?

**Comment by avsm** [comment]
once those two are fixed, the reverse dependencies will also be built by the CI.

**Comment by diml** [comment]
We looked at this with @hhugo. `async_js` supersedes `js_of_ocaml.{async,ppx_driver}`, so we disabled the building of `js_of_ocaml.{async,ppx_driver}` in 2.8.4. Hopefully only the JS packages are using these

**Comment by avsm** [comment]
sgtm, I'm not aware of any other users of Async+JS

**Comment by avsm** [comment]
Gist bundle of the build failures of reverse dependencies from this PR are at: https://gist.github.com/avsm/a2c6ad9753c2a1450ada07c8e816b778

**Comment by avsm** [comment]
Closing this PR temporarily so that it can skip the CI and not block it, will reopen when the upper bounds from the failures above are in place.

**Comment by avsm** [comment]
Running this through [scry](https://github.com/ocamllabs/ocaml-scry) has this summary of failures. Lots of ppx_deriving errors:

| Package | Failure Description |
| -- | -- |
| amqp-client.txt | FAIL (error in dependency "amqp-client.1.0.4" (compilation error at "amqp_thread.ml", line 1)) |
| arakoon.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| bap-future.txt | FAIL (error in dependency "" (compilation error at "lib/bap_future/bap_future.ml", line 116, characters 30-158)) |
| bap-ida.txt | FAIL (unknown) |
| bap-server.txt | FAIL (unknown) |
| bap-std.txt | FAIL (unknown) |
| bap-veri.txt | FAIL (error in dependency "" (compilation error at "lib/veri_rule.mli", line 9, characters 8-15)) |
| batsh.txt | FAIL (unknown) |
| biocaml.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| bookaml.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| camlhighlight.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| capnp.txt | FAIL (error in dependency "" (compilation error at "./make_includes.ml", line 1)) |
| caravan.txt | FAIL (error in dependency "" (compilation error at "src/caravan/caravan.mli", line 1)) |
| cfstream.txt | FAIL (error in dependency "" (compilation error at "app/cfstream_test.ml", line 3, characters 5-20)) |
| charrua-core.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| cohttp.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| conduit.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| core-lwt.txt | FAIL (error in dependency "" (compilation error at "lib/core_lwt_basic.ml", line 4, characters 19-165)) |
| datakit-ci.txt | FAIL (unknown) |
| datakit-server.txt | FAIL (error in dependency "" (compilation error at "lib/protocol_9p_types.ml", line 152, characters 2-102)) |
| dns-forward.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| dockerfile.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| ezirmin.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| future.txt | FAIL (error in dependency "" (compilation error at "lib/unix/future.ml", line 9, characters 21-136)) |
| get_line.txt | FAIL (unknown) |
| git.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| gnuplot.txt | FAIL (error in dependency "" (compilation error at "lib/gnuplot.ml", line 60, characters 67-71)) |
| graphlib.txt | FAIL (error in dependency "" (compilation error at "lib/regular/regular_bytes.ml", line 64, characters 18-21)) |
| http_router.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| ipaddr.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| key-parsers.txt | FAIL (error in dependency "" (compilation error at "src/kp_asn1.ml", line 90, characters 9-12)) |
| libsvm.txt | FAIL (error in dependency "" (missing findlib constraint "sexplib.syntax ")) |
| mariadb.txt | FAIL (unknown) |
| mirage-http.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| mirage-net-xen.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| nbd.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| nocrypto.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| oclaunch.txt | FAIL (error in dependency "" (compilation error at "src/list_rc.ml", line 61, characters 53-66)) |
| opam-sync-github-prs.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| opass.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| opium.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| otr.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| planck.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| ppx_fun.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_tools.metaquot")) |
| ppx_jsobject_conv.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| prob-cache.txt | FAIL (error in dependency "" (compilation error at "mylogging.ml", line 34, characters 65-66)) |
| protocol-9p.txt | FAIL (error in dependency "" (compilation error at "lib/protocol_9p_types.ml", line 152, characters 2-102)) |
| qcow-format.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| qcow.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| regular.txt | FAIL (error in dependency "" (compilation error at "lib/regular/regular_bytes.ml", line 64, characters 18-21)) |
| riakc_ppx.txt | FAIL (error in dependency "" (compilation error at "mylogging.ml", line 34, characters 65-66)) |
| rubytt.txt | FAIL (error in dependency "rubytt.0.1" (compilation error at "type.ml", line 238, characters 38-202)) |
| sociaml-vcard.txt | FAIL (unknown) |
| tensorflow.txt | FAIL (error in dependency "" (compilation error at "caml_startup", line 1)) |
| text-tags.txt | FAIL (error in dependency "" (compilation error at "lib/text_tags/text_tags.ml", line 176, characters 2-29)) |
| tidy.txt | FAIL (error in dependency "" (no external dependency "tidy.h")) |
| tls.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| tlstunnel.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| unmagic.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| uri.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| vchan.txt | FAIL (error in dependency "" (missing findlib constraint "ppx_deriving ")) |
| vrt.txt | FAIL (error in dependency "" (missing ocamlfind dependency "sexplib.syntax")) |
| wamp.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |
| x509.txt | FAIL (error in dependency "" (missing ocamlfind dependency "ppx_deriving")) |

**Comment by avsm** [comment]
from a chat with @diml, the ppx_deriving dependency failures are due to the newer versions of (e.g.) ppx_sexp_conv not having a hard dependency on ppx_deriving anymore.  Instead, if a package uses ppx_sexp_conv through ocamlfind, it needs to explicitly depend on ppx_deriving now as well.  A package that uses Jbuilder or ppx_driver can skip this dependency.

@diml: this is probably worth an entry in the CHANGELOG file similar to the camlp4 sexplib change...?

**Comment by diml** [comment]
Indeed, I'll add an entry

**Pull request closed without merging**